### ISSUE TITLE
[Unity] CustomAvatar.dll will not be loaded due to errors

### ISSUE THREAD

**Issue by Karma-Fox** [nicoco007/BeatSaberCustomAvatars#122]
**Before doing anything else, please attach your latest log to this issue. It should be located at `Logs\_latest.log` in your Beat Saber installation's folder.**
Not a Beat Saber issue

**Desktop (please complete the following information):**
- Operating system: Windows 10
- Headset and controllers: Oculus Rift CV1

**Describe the bug**
When trying to create a custom avatar in unity, CustomAvatar.dll causes the following error;

```
Assembly 'Assets/CustomAvatar.dll' will not be loaded due to errors:
Unable to resolve reference 'Zenject-usage'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'IPA.Loader'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference '0Harmony'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Zenject'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Main'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'HMUI'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'HMLib'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Newtonsoft.Json'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'UnityEngine.UI'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Unity.TextMeshPro'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'BSML'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'DynamicOpenVR'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Rendering'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'HMRendering'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'BeatmapCore'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'GameplayCore'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'BGNet'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'VRUI'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Polyglot'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
```

I have tried to use this dll in multiple versions of Unity including 2018.4.11f1, 2019.4.31f1, and 2019.4.28f1. I have also tried to use it in a new project and I have even attempted to use the included Unity Project package provided here. Lastly, I have even attempted to delete and replace the CustomAvatar.dll pack to no avail. No matter what I do, this dll will not work for me.

**Steps to reproduce**
1. Install Unity 2019.4.28f1
2. Download the 5.2.9 Unity Project from Releases
3. Open BeatSaberCustomAvatars Project in Unity

**Expected behavior**
No Errors, period.

**Screenshots**
![image](https://user-images.githubusercontent.com/75238752/135941658-b2ea9ee9-eb17-4383-9b40-716ba8809dfc.png)

**Comment by nicoco007** [comment]
Looks like you're trying to use the wrong DLL. You should be using the one from [BeatSaberCustomAvatars-v5.2.9-Editor.zip](https://github.com/nicoco007/BeatSaberCustomAvatars/releases/download/v5.2.9/BeatSaberCustomAvatars-v5.2.9-Editor.zip), not the regular plugin from DEBUG/RELEASE. Also, you should be using either Unity 2019.3.15f1 if you plan on uploading to ModelSaber or any 2019.4 version equal to or under 2019.4.28f1 otherwise when opening the project since that's the version Beat Saber is currently running on.

**Comment by Karma-Fox** [comment]
2019.4.28f1 is what I was using. I will give the Editor DLL a try though I was using the one that was included with BeatSaberCustomAvatars-v5.2.9-UnityProject.zip

**Comment by Karma-Fox** [comment]
That DLL seemed to have done the trick. Thank you
